### ISSUE TITLE
add back blame-on-current-line behavior to git-extras

### ISSUE THREAD

**Issue by sqs** [sourcegraph/sourcegraph#1552]
Sourcegraph used to show blame info for the selected line. Now it only lets you toggle blame on/off for the entire file, and it shows the blame info for the first line of each hunk. Many people liked the previous behavior, and we want to bring it back.

Blocked on #2026 to add the extension APIs necessary for this.

**Comment by sqs** [comment]
Waiting for @lguychard to add the following to the extension API, then I will implement this in the git-extras extension:

- ability to listen for changes to active text editor
- ability to listen for changes to `TextEditor#selections`

**Comment by lguychard** [comment]
> ability to listen for changes to `TextEditor#selections`

@sqs investigated this, and it's a bigger change than I expected, notably because there is currently no support for `TextEditor#selections` on [code hosts](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/browser/src/libs/code_intelligence/code_intelligence.tsx#L519) (selections only work in the web app).

How about doing this through `registerHoverProvider` for 3.0?

**Comment by sqs** [comment]
@lguychard It’s OK for there to not be selections from code hosts yet. Bringing back the Git blame on selected line feature for Sourcegraph is good enough.

**Comment by sqs** [comment]
Filed the extension APIs that are necessary for this in #2026, to keep this purely about updating the git-extras extension to use these new APIs.

**Comment by sqs** [comment]
Closing in favor of #1113
